### ISSUE TITLE
Atomic memory function can only be used for shader storage block member or shared variable.

### ISSUE THREAD

**Issue by ZhiqianXia** [KhronosGroup/glslang#2642]
Hi , 
The spec says :
A shader will **fail** to compile if the value passed to the mem argument of an atomic memory function does not correspond to a buffer or shared variable.  But the following shader didn't get an error from glslang. 

**shader**:
```
#version 310 es
layout(local_size_x = 1) in;
int y;
void main() {
  atomicAdd(y, 2);   // uint atomicAdd (inout uint mem, uint data)
}
```
ps: i'll submit the PR later.

**Comment by ZhiqianXia** [comment]
https://github.com/KhronosGroup/glslang/pull/2643

**Comment by greg-lunarg** [comment]
Fixed by #2643
